src/charm.py

```diff
@@ -99,86 +102,142 @@ def _configure_pebble(self) -> None:
             self._container.add_layer(self._container_name, layer, combine=True)
             self._container.restart(self._service_name)
 
-    def _configure_upf_information(self):
+    def _configure_upf_information(self) -> None:
         """The `fiveg_n4` relation is not mandatory.
 
-        If it exists it must contain the UPF hostname and port.
+        If it exists, config files are generated and pushed to the workload.
         """
-        if self.model.relations.get(FIVEG_N4_RELATION_NAME):
-            if not self._get_upf_hostname() or not self._get_upf_port():
-                logger.warning("Invalid information in %s integration", FIVEG_N4_RELATION_NAME)
+        if not self.model.relations.get(FIVEG_N4_RELATION_NAME):
+            logger.warning("Relation %s not available", FIVEG_N4_RELATION_NAME)
+            return
+        upf_existing_content = self._get_existing_config_file(path=UPF_CONFIG_PATH)
+        upf_config_content = self._get_upf_config()
+        if not upf_config_content:
+            logger.warning("UPF config file is invalid")
```

Comment:
This should probably be an error, we need this file to be valid for the NMS to work